### PR TITLE
basic logging to help debug 404 client errors

### PR DESCRIPTION
**Pull request by pnepywoda**
@uschi2000

### CODE REVIEW

**Comment by pnepywoda** [comment]
logs to DEBUG on slf4j logger for FeignClientFactory so clients can activate/deactivate via yml

**Comment by pnepywoda** [comment]
@markelliot

File: http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java [hunk 1]
```diff
@@ -104,6 +106,8 @@ public static FeignClientFactory of(
                 .errorDecoder(errorDecoder)
                 .client(clientSupplier.apply(sslSocketFactory))
                 .options(options)
+                .logger(new Slf4jLogger(FeignClientFactory.class))
+                .logLevel(Level.BASIC)
```
Comment (uschi2000) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java:110]: I assume you can override this using the standard mechanism provided by your slf4j implementation?

File: http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java [hunk 2]
```diff
@@ -124,6 +128,8 @@ public static FeignClientFactory of(
                 .client(client)
                 .retryer(target)
                 .options(options)
+                .logger(new Slf4jLogger(FeignClientFactory.class))
```
Comment (markelliot) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java]: I'd prefer we use a different string than FeignClientFactory, and should probably encode it as a constant.

Comment (pnepywoda) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java]: Ok, using FeignClients

File: http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java (hunk 1 above)
Comment (pnepywoda) [inline http-clients/src/main/java/com/palantir/remoting/http/FeignClientFactory.java:110]: Yeah, by default it logs nothing cause it logs on slf4j's DEBUG by default

**Comment by uschi2000** [comment]
:+1: 
anything left, @markelliot ?

**Comment by markelliot** [comment]
lgtm

**Pull request merged**